### PR TITLE
Icons components

### PR DESCRIPTION
**Pull request by poulch**
Fixes https://github.com/saleor/macaw-ui/issues/197

### What was added:
- SVG icons with size variants and colour based on the theme
- All SVGs have been optimized

### CODE REVIEW

File: src/components/icons/SVGWrapper/SVGWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  children: ReactNode;
+};
+
+export const SVGWrapper = forwardRef<SVGSVGElement, SVGWrapperProps>(
+  (
+    { className, size, viewBox = "0 0 24 24", color = "black", children },
+    ref
+  ) => {
+    return (
+      <svg
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/SVGWrapper.tsx:17]: **suggestion**: What about using here `<Box/>`?

I think you can use [<Box/>](https://github.com/saleor/macaw-ui/blob/canary/src/components/Box/Box.tsx) component here and pass color as a prop

File: src/components/icons/SVGWrapper/SVGWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { ReactNode, forwardRef } from "react";
+import { svgWrapper, SCVWrapperVaraints } from "./SVGWrapper.css";
+
+export type SVGWrapperProps = SCVWrapperVaraints & {
+  className?: string;
+  viewBox?: string;
+  color?: string;
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/SVGWrapper.tsx:7]: **question**: shouldn't it take only values from `Sprinkles['color']`?

File: src/components/icons/SVGWrapper/SVGWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  children: ReactNode;
+};
+
+export const SVGWrapper = forwardRef<SVGSVGElement, SVGWrapperProps>(
+  (
+    { className, size, viewBox = "0 0 24 24", color = "black", children },
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/SVGWrapper.tsx:13]:
```suggestion
    { className, size, viewBox = "0 0 24 24", color = "currentColor", children },
```

File: src/components/icons/SVGWrapper/createSVGWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { ReactNode } from "react";
+import { SVGWrapper, SVGWrapperProps } from "./SVGWrapper";
+
+export const createSVGWrapper = (node: ReactNode) => {
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/createSVGWrapper.tsx:4]: **question**: do we need this wrapper?

Can't we use `<SVGWrapper/>` as component?

File: src/components/index.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export * from "./Box";
+export * from "./icons";
```
Comment (krzysztofzuraw) [inline src/components/index.ts:2]: **suggestion**: to be consistent in naming maybe it should be capitalized (`./Icons`)?

File: src/components/icons/icons.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const macawIcons = Object.values(icons);
+
+export default {
+  title: "components/Icons",
+};
+
+export const Default = () =>
+  ["Small", "Medium", "Large"].map((size) => (
+    <>
+      <h3>{size}</h3>
```
Comment (krzysztofzuraw) [inline src/components/icons/icons.stories.tsx:13]:
```suggestion
      <Text as="h3">{size}</Text>
```

File: src/components/icons/icons.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      <div
```
Comment (krzysztofzuraw) [inline src/components/icons/icons.stories.tsx:14]:
```suggestion
      <Box
```

What about using `<Box/>` here as well?

File: src/components/icons/icons.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        }}
+      >
+        {macawIcons.map((Icon, idx) => (
+          <div
+            style={{
+              margin: "10px",
+            }}
+            key={idx}
+          >
+            <Icon size={size.toLowerCase() as keyof SCVWrapperVaraints} />
```
Comment (krzysztofzuraw) [inline src/components/icons/icons.stories.tsx:29]: **suggestion**: move those size to their own variable

You can have `const sizes=['small','medium','large'] as const` and do not cast strings to `SCVWrapperVaraints` here

File: src/components/icons/icons.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      <div
+        style={{
+          display: "flex",
+          alignItems: "center",
+          flexDirection: "row",
+          flexWrap: "wrap",
+        }}
+      >
+        {macawIcons.map((Icon, idx) => (
+          <div
```
Comment (krzysztofzuraw) [inline src/components/icons/icons.stories.tsx:23]: **suggestion**: What about using `<Box/>` here as well?

File: src/components/icons/SVGWrapper/SVGWrapper.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    ref
+  ) => {
+    return (
+      <svg
+        ref={ref}
+        className={`${svgWrapper({ size })} ${className}`}
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/SVGWrapper.tsx:19]: **suggestion**: what about using `clsx` utility here? 

We could use [clsx](https://www.npmjs.com/package/clsx) to concatenate `classNames`

File: src/components/icons/SVGWrapper/SVGWrapper.css.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import { recipe, RecipeVariants } from "@vanilla-extract/recipes";
+
+import { sprinkles } from "~/theme";
+
+export const svgWrapper = recipe({
+  base: [sprinkles({ color: "neutralTextPrimary" })],
```
Comment (krzysztofzuraw) [inline src/components/icons/SVGWrapper/SVGWrapper.css.ts:6]: **question**: do you need color here?

If you have `color = "black"` passed as a prop in [SVGWrapper.tsx](https://github.com/saleor/macaw-ui/pull/205/files#diff-d4c1a555b9f1c77a8f5b85587477ca4b31de1a7df8b1fe5fe2aaaab02368c483) do you need color defined here as well?

File: src/components/icons/SVGWrapper/createSVGWrapper.tsx (hunk 4 above)
Comment (poulch) [reply src/components/icons/SVGWrapper/createSVGWrapper.tsx:4]: With createSVGWrapper we don't have to import types for props in each icon component

File: src/components/index.ts (hunk 5 above)
Comment (poulch) [reply src/components/index.ts:2]: Do we want ot use capitilize icons in import like this 
`@saleor/macaw-ui/components/icons/ArrowUpIcon` ?

File: src/components/icons/SVGWrapper/SVGWrapper.tsx (hunk 10 above)
Comment (poulch) [reply src/components/icons/SVGWrapper/SVGWrapper.tsx:19]: IMO this is enough `className={[svgWrapper({ size }), className].join(" ")}`

File: src/components/icons/SVGWrapper/SVGWrapper.tsx (hunk 1 above)
Comment (poulch) [reply src/components/icons/SVGWrapper/SVGWrapper.tsx:17]: In this case I don't think so as Box doesn't support svg props as viewBox

**Review by krzysztofzuraw — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

File: .npmrc [hunk 12]
```diff
@@ -1 +1,2 @@
 legacy-peer-deps=true
+shamefully-hoist=true
```
[the remark this answers is not in the document]
Comment (poulch) [reply .npmrc:2]: Try to remove this line, remove node_modules, install node_modules and run project :)

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**